### PR TITLE
Fix conda publish workflow

### PR DESCRIPTION
**Pull request by fdiblen**
### **Patch release: 0.3.4**

Fixes the issues of Anaconda publish workflow.

### Changed
- Fix verify step in conda publish workflow

### CODE REVIEW

File: .github/workflows/conda_publish.yml [hunk 1]
```diff
@@ -74,6 +74,13 @@ jobs:
         shell: bash -l {0}
         run: |
           conda install --channel bioconda --channel conda-forge --channel nlesc matchms
+      - name: Remove the package from Anaconda Cloud due to installation failure
+        if: ${{ failure() }}
```
Comment (sverhoeven) [inline .github/workflows/conda_publish.yml]: Normally you would create a new release with a fix. You are not rolling back the release on GH and Zenodo so I would either remove the release from Anaconda, Zenodo and GH or leave them all.

Normally the rest of the steps are not performed when the prev step fails. Is there a build which actually removed the anaconda package?

Comment (fdiblen) [reply .github/workflows/conda_publish.yml]: We need to have a short chat.

**Review by sverhoeven — APPROVED** [review]
Easy to review now, looks ok to me.

File: .github/workflows/conda_publish.yml (hunk 1 above)
Comment (fdiblen) [reply .github/workflows/conda_publish.yml]: done

**Comment by sverhoeven** [comment]
:clap:  Congrats with successful [publish job](https://github.com/matchms/matchms/actions/runs/118814803).

**Pull request merged**